praw/__init__.py

```diff
@@ -1689,16 +1691,19 @@ def upload_image(self, subreddit, image_path, name=None, header=False):
             size = os.path.getsize(image.name)
             if size < MIN_IMAGE_SIZE:
                 raise errors.ClientException('`image` is not a valid image')
-            elif size > MAX_IMAGE_SIZE:
+            if size > MAX_IMAGE_SIZE:
                 raise errors.ClientException('`image` is too big. Max: {0} '
                                              'bytes'.format(MAX_IMAGE_SIZE))
-            first_bytes = image.read(MIN_IMAGE_SIZE)
+            first_bytes = image.read(MIN_PNG_SIZE)
             image.seek(0)
-            if first_bytes.startswith(JPEG_HEADER):
-                image_type = 'jpg'
-            elif first_bytes.startswith(PNG_HEADER):
+            if first_bytes.startswith(PNG_HEADER):
                 image_type = 'png'
             else:
+                first_bytes = image.read(MIN_JPEG_SIZE)
+                image.seek(0)
+                if first_bytes.startswith(JPEG_HEADER):
+                    image_type = 'jpg'
+            if image_type is None:
```

Comment:
https://gist.github.com/Andygmb/093db0962e5f7bd39fde

Is the above OK? I can make a new PR (or add a new commit to this one, unsure of which is the standard here)